src/app/core/services/repo-creator.service.ts

```diff
@@ -18,9 +20,43 @@ export const BUG_REPORTING_INVALID_ROLE =
 export class RepoCreatorService {
   constructor(
     private githubService: GithubService,
-    private userService: UserService
+    private userService: UserService,
+    private phaseFixConfirmationDialog: MatDialog
   ) {}
 
+  /**
+   * Checks if the current repository has been created and if not, checks
+   * if the current phase match the requirements to make a request for
+   * repo creation permissions.
+   * @param currentPhase the current phase of the session.
+   * @param phaseRepo the name of the specified repository.
+   */
+  public requestRepoCreationPermissions(currentPhase: Phase, phaseRepo: string):
+    UnaryFunction<Observable<boolean>, Observable<boolean | null>> {
+    return pipe(
+      flatMap((isSessionAvailable: boolean) => {
+        if (!isSessionAvailable && currentPhase === Phase.phaseBugReporting) {
+          return this.openSessionFixConfirmation(phaseRepo);
+        } else {
+          return of(null);
+        }
+      })
+    );
+  }
+
+  /**
+   * Launches the SessionFixConfirmation Dialog.
+   * @param phaseRepo the name of the specified repository.
+   * @return Observable<boolean> - Representing user's permission grant.
+   */
+  private openSessionFixConfirmation(phaseRepo: string): Observable<boolean> {
+    const dialogRef: MatDialogRef<SessionFixConfirmationComponent> =
```

Comment:
Let's rename the `SessionFixConfirmationComponent` --> `RepoCreationConfirmationComponent` too.
In my view, "RepoCreation" has a more obvious meaning than "SessionFix".
What do you think?

This renaming can be done separately in another PR.

Reply:
Yes it would be better to do so! 

I think we can do this in a separate PR as well :smile: